app/models/course/assessment/submission.rb

```diff
@@ -123,4 +125,10 @@ def auto_grade_submission
       auto_grade!
     end
   end
+
+  # Validate that the submission creator is the same user as the course_user in the associated
+  # experience_points_record.
+  def validate_consistent_user
+    errors.add(:base, :inconsistent_user) unless course_user && course_user.user == creator
```

Comment:
the first argument should be the object property that contains the validation error. In this case, it should be `:experience_points_record` (IIRC)